### PR TITLE
Change adapter discovery behavior on Windows

### PR DESCRIPTION
**Pull request by kswiecicki**

### CODE REVIEW

**Comment by pbalcer** [comment]
please describe the reasoning why we are doing this change in the commit message.

**Comment by kswiecicki** [comment]
> please describe the reasoning why we are doing this change in the commit message.

I've added a brief message saying that it is an SDL requirement for Windows DLL search.

**Comment by DamianDuy** [comment]
LGTM

File: scripts/core/INTRO.rst [hunk 1]
```diff
@@ -234,6 +234,28 @@ An example of an environment variable for setting up the null adapter library wi
...
+---------------------
+UR is capable of discovering adapter libraries in the following ways in the listed order:
+
+  - Check paths to the adapters set in `UR_ADAPTERS_FORCE_LOAD` environment variable.
+
+    + All other adapter discovery methods are disabled when this environment variable is used.
+
+  - Search directories specified in `UR_ADAPTERS_SEARCH_PATH` environment variable.
+
+  - Leave adapter discovery for the OS.
```
Comment (pbalcer) [inline scripts/core/INTRO.rst:247]: - If on Linux, use the shared library discovery mechanism (see `ld.so(8)` for details).

File: scripts/core/INTRO.rst [hunk 2]
```diff
@@ -234,6 +234,28 @@ An example of an environment variable for setting up the null adapter library wi
 
   UR_LOG_NULL="level:warning;output:stdout"
 
+Adapter Discovery
+---------------------
+UR is capable of discovering adapter libraries in the following ways in the listed order:
+
+  - Check paths to the adapters set in `UR_ADAPTERS_FORCE_LOAD` environment variable.
```
Comment (pbalcer) [inline scripts/core/INTRO.rst:241]: Search in

File: scripts/core/INTRO.rst [hunk 3]
```diff
@@ -234,6 +234,28 @@ An example of an environment variable for setting up the null adapter library wi
...
+
+    + All other adapter discovery methods are disabled when this environment variable is used.
+
+  - Search directories specified in `UR_ADAPTERS_SEARCH_PATH` environment variable.
```
Comment (pbalcer) [inline scripts/core/INTRO.rst:245]: Search in

File: scripts/core/INTRO.rst [hunk 4]
```diff
@@ -234,6 +234,28 @@ An example of an environment variable for setting up the null adapter library wi
...
+
+  - Leave adapter discovery for the OS.
+
+    + This method is disabled on Windows.
+
+  - Explore the directory at the UR loader location.
```
Comment (pbalcer) [inline scripts/core/INTRO.rst:251]: Search in

File: scripts/core/INTRO.rst (hunk 1 above)
Comment (kswiecicki) [reply scripts/core/INTRO.rst:247]: Done.

**Pull request merged**